Move tour to overview

BillWagner opened the pull request:
Fixes #3276 (finally)

- Update the TOC to move the "tour of C#" under the overview node.
- Remove the existing "get started" index. It was an old landing page.
- Update the remaining "Introduction to C# and .NET Framework" article for .NET 5 and architectural changes.

[Internal review link](https://review.docs.microsoft.com/en-us/dotnet/csharp/getting-started/?branch=pr-en-us-21066)

BillWagner committed:
finalize Get Started experience

Move the "Tour of C#" under the overview section in the TOC)
Remove the old Get Started index, which was an old landing page. Replace it with the current "introduction to C# and .NET".
Finally, udpate the introduction to C# and .NET to .NET 5 concepts.

BillWagner committed:
Update language and grammar.

Youssef1313 commented on docs/csharp/getting-started/index.md:
```diff
@@ -1,51 +1,47 @@
-  - "Visual C#, getting started"
-  - "getting started, Visual C#"
-ms.date: 04/05/2019
+title: Get Started - an introduction to the C# language and .NET"
+description: Learn the basics of C# and .NET. Get an overview of the C# language and .NET ecosystem.
+ms.date: 10/13/2020
```
Are the helpviewer_keywords intentionally removed?
I assume they have something to do with SEO

BillWagner replied:
They were intentionally removed. They were historically used for search in the offline CHM versions of Visual Studio help. They aren't needed any more.

BillWagner committed:
update tour breadcrumb

IEvangelist commented on docs/csharp/getting-started/index.md:
```diff
@@ -1,51 +1,47 @@
+## .NET Platform Architecture
 
-   This topic shows you how to create and run a simple Hello World application with Visual Studio Code and .NET Core.
+C# programs run on .NET, a virtual execution system called the common language runtime (CLR) and a unified set of class libraries. The CLR is the commercial implementation by Microsoft of the common language infrastructure (CLI), an international standard. The CLIis the basis for creating execution and development environments in which languages and libraries work together seamlessly.
```
```suggestion
C# programs run on .NET, a virtual execution system called the common language runtime (CLR) and a unified set of class libraries. The CLR is the commercial implementation by Microsoft of the common language infrastructure (CLI), an international standard. The CLI is the basis for creating execution and development environments in which languages and libraries work together seamlessly.
```

IEvangelist commented on docs/csharp/getting-started/index.md:
```diff
@@ -1,51 +1,47 @@
+Language interoperability is a key feature of .NET. IL code produced by the C# compiler conforms to the Common Type Specification (CTS). IL code generated from C# can interact with code that was generated from the .NET versions of F#, Visual Basic, C++, or any of more than 20 other CTS-compliant languages. A single assembly may contain multiple modules written in different .NET languages, and the ...
 
-- [C# Reference](../language-reference/index.md)
+In addition to the run time services, .NET also includes an extensive library of over 4000 classes. The .NET library is organized into namespaces that provide a wide variety of useful functionality for everything from file input and output to string manipulation to XML parsing, to web application frameworks to Windows Forms controls. The typical C# application uses the .NET class library extensiv ...
```
```suggestion
In addition to the run time services, .NET also includes an extensive library of over 4,000 classes. The .NET library is organized into namespaces that provide a wide variety of useful functionality for everything from file input and output to string manipulation to XML parsing, to web application frameworks to Windows Forms controls. The typical C# application uses the .NET class library extensively to handle common "plumbing" chores.
```

Is 4,000 classes accurate - I'd imagine that it's WAY more now? Also, you mention the ".NET library" here as if there is only one, which one are you referring to - I'm confused by that? Do you mean the runtime? If so the, there are multiple implementations...I'm not sure how to update this. Maybe a link to the library?

IEvangelist reviewed (approved):
I had a few questions and suggestions. Looks good to me though, :shipit: when you're ready.

BillWagner committed:
Apply suggestions from code review

Co-authored-by: David Pine <david.pine@microsoft.com>

BillWagner committed:
rework description of libraries and app models.

4000 is an outdated number of classes.
The more important point is the support for a variety of app models.
